### ISSUE TITLE
IMeterFactory.Create triggers false positive on IDisposable analysis

### ISSUE THREAD

**Issue by noahfalk** [dotnet/roslyn-analyzers#6912]
### Analyzer

**Diagnostic ID**: [CA2000](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2000): `Dispose objects before losing scope`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 8 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

### Describe the bug

.NET 8 adds a new API [IMeterFactory.Create](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.metrics.meterfactoryextensions.create) which returns instances of type `Meter`. Although the returned Meter object implements IDisposable, the IMeterFactory retains responsibility for invoking Dispose() when the factory is disposed. However the analyzer believes that the caller has the responsibility to call Dispose() and triggers a warning incorrectly.

### Steps To Reproduce

1. Create a new console project in VS
2. Add this code into it (this is taken from the [docs](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/metrics-instrumentation#get-a-meter-via-dependency-injection)):

```C#
public class HatCoMetrics
{
    private readonly Counter<int> _hatsSold;

    public HatCoMetrics(IMeterFactory meterFactory)
    {
        var meter = meterFactory.Create("HatCo.Store");
        _hatsSold = meter.CreateCounter<int>("hatco.store.hats_sold");
    }

    public void HatsSold(int quantity)
    {
        _hatsSold.Add(quantity);
    }
}
```

3. Change the analysis rules for the project to 'Latest all'
4. Build

### Expected behavior

No warning CA2000

### Actual behavior

Warning CA2000 is generated by the build

**Comment by mavasani** [comment]
CA2000 uses a heuristic to identify factory methods that create a disposable object that transfer dispose ownership, and static methods named `Create` that return an IDisposable are one such set. You should add source suppressions to suppress these for now. The core issue can only be resolved if we implement #1617 and https://github.com/dotnet/runtime/issues/29631

Closing this as not-planned, but the above two issues track the core feature request here.
